sound/soc/sof/ops.h

```diff
@@ -387,10 +387,47 @@ static inline int snd_sof_dma_trace_release(struct snd_sof_dev *sdev)
 
 static inline int snd_sof_dma_trace_trigger(struct snd_sof_dev *sdev, int cmd)
 {
-	if (sof_ops(sdev)->trace_trigger)
-		return sof_ops(sdev)->trace_trigger(sdev, cmd);
+	int ret = 0;
 
-	return 0;
+	if (!sdev->dtrace_is_supported)
+		return 0;
+
+	/* FIXME: return 0 or return -EINVAL ? */
+	if (!sof_ops(sdev)->trace_trigger)
+		return 0;
+
+	switch (cmd) {
+	case SNDRV_PCM_TRIGGER_START:
+		/* FIXME: return 0 when dtrace_is_enabled? */
+		if (sdev->dtrace_is_enabled)
+			return 0;
+		if (!sdev->dma_trace_pages)
+			return -EINVAL;
+		ret = sof_ops(sdev)->trace_trigger(sdev, cmd);
+		if (ret < 0) {
+			dev_err(sdev->dev,
+				"error: snd_sof_dma_trace_trigger: start: %d\n", ret);
```

Comment:
remove err prefix

Reply:
ACK. Yes, removing the "error" is better.

Reply:
hold on, although not a fan, @ranj063  I think we still generally add "error: "to dev_errs . @plbossart @ujfalusi  have we changed policy on this?

Reply:
@kv2019i we want to remove "error:" since it makes no sense, but we will do it gradually through new changes to avoid backport issues. So were we should remove the "error:" on line 409.